modules/openxr/editor/openxr_action_map_editor.cpp

```diff
@@ -208,7 +208,7 @@ void OpenXRActionMapEditor::_on_remove_action_set(Object *p_action_set_editor) {
 void OpenXRActionMapEditor::_on_action_removed() {
 	for (int i = 0; i < tabs->get_tab_count(); i++) {
 		// First tab won't be an interaction profile editor, but being thorough..
-		OpenXRInteractionProfileEditorBase *interaction_profile_editor = static_cast<OpenXRInteractionProfileEditorBase *>(tabs->get_tab_control(i));
+		OpenXRInteractionProfileEditorBase *interaction_profile_editor = Object::cast_to<OpenXRInteractionProfileEditorBase>(tabs->get_tab_control(i));
 		if (interaction_profile_editor) {
```

Comment:
This seems to be dead code, but I left it in as it may still be relevant for some future plan?

Reply:
So not dead code, but maybe overkill until we actually start adding more classes in for the different tabs. Basically `Object::cast_to<T>` returns `nullptr` for the first tab as it is not a subclass of `OpenXRInteractionProfileEditorBase` and the logic is skipped for that tab..

Reply:
It's inside of `Object::cast_to<T>`, it performs a dynamic cast which as far as I've been told, will return `nullptr` if the object can't be cast to that type.

Reply:
though admittedly it should throw an exception at this point too so I might be assuming behavior here that is wrong, hmmm. weird.

Reply:
Ah reading it wrong: `If the cast is successful, dynamic_cast returns a value of type new-type. If the cast fails and new-type is a pointer type, it returns a null pointer of that type. If the cast fails and new-type is a reference type, it throws an exception that matches a handler of type [std::bad_cast](https://en.cppreference.com/w/cpp/types/bad_cast).`

The exception is throws if its a reference type. If its a pointer it will indeed just return null.

Reply:
Ok, so there is a small side effect :)

That said, I still don't get what the whole `_on_action_removed` is supposed to do in its current state. It does not seem to have any desired side effects during regular operation?

Reply:
Oh hang on, sorry Dirk, I HATE these diff checks, I'm completely barking up the wrong tree here

Reply:
Yeah there is no code there is there :P

There should be, if I recall correctly it should be calling `remove_all_bindings_for_action` there, I wonder how that got lost...